File: hackhive-backend/src/main/java/com/hackhive/submission/service/ProjectSubmissionEvaluationService.java
package com.hackhive.submission.service;

import com.hackhive.submission.dto.request.CreateUpdateEvaluationRequest;
import com.hackhive.submission.dto.response.ProjectSubmissionEvaluationResponse;

public interface ProjectSubmissionEvaluationService {

    ProjectSubmissionEvaluationResponse saveEvaluationDraft(Long submissionId, CreateUpdateEvaluationRequest request);

    ProjectSubmissionEvaluationResponse getEvaluationBySubmissionId(Long submissionId);

    ProjectSubmissionEvaluationResponse updateEvaluationDraft(Long evaluationId, CreateUpdateEvaluationRequest request);

    ProjectSubmissionEvaluationResponse finalizeEvaluation(Long evaluationId);
}

File: hackhive-frontend/src/services/projectSubmissionEvaluationService.js
import api from "../api/axios";

async function request(promise) {
    const response = await promise;
    return response.data?.data ?? response.data;
}

export const projectSubmissionEvaluationService = {
    getEvaluationBySubmissionId: (submissionId) =>
        request(api.get(`/project-submission-evaluations/submissions/${submissionId}`)),

    saveEvaluationDraft: (submissionId, data) =>
        request(api.post(`/project-submission-evaluations/submissions/${submissionId}`, data)),

    updateEvaluationDraft: (evaluationId, data) =>
        request(api.put(`/project-submission-evaluations/${evaluationId}`, data)),

    finalizeEvaluation: (evaluationId) =>
        request(api.post(`/project-submission-evaluations/${evaluationId}/finalize`)),
};

export default projectSubmissionEvaluationService;
